File: php/types/strings/strings.php
#!/usr/bin/php8.0
<?php

$s = "hello";
print("string \"hello\": $s (" . gettype($s) . ")\n");

$s = 'hello';
print("string 'hello': $s (" . gettype($s) . ")\n");

print "length: " . strlen($s) . "\n\n";

$cite = "Stat rosa pristina nomine, nomina nuda tenemus.";
print "cite: $cite\n";
print "words quantity: " . str_word_count($cite) . "\n\n";

$s1 = "abc";
$s2 = "def";
$s3 = $s1 . $s2;
print "$s1 . $s2 = $s3\n\n";

File: c++/types/strings/strings.cpp
#include <iostream>
#include <string>
#include <typeinfo>

int main()
{
  std::string s = "hello";
  std::cout << "\"hello\": " << s << " (" << typeid(s).name() << ")\n";

  std::cout << "length: " << s.length() << "\n\n";

  std::string cite = "Stat rosa pristina nomine, nomina nuda tenemus.";
  std::cout << "cite: " << cite << "\n\n";

  std::string s1 = "abc";
  std::string s2 = "def";
  std::string s3 = s1 + s2;
  std::cout << s1 << " + " << s2 << " = " << s3 << "\n\n";
}
